pyiron_atomistics/dft/job/generic.py

```diff
@@ -501,6 +515,54 @@ def get_density_of_states(self, sigma=0.1, shift_by_fermi_energy=True, grid=None
         hist *= 2 / len(eigen_values) / np.sqrt(2 * np.pi * sigma**2)
         return {"grid": grid, "dos": hist}
 
+class DFTInput():
+    
+    def __init__(self):
+        self.structure = None
+        self._idx_user_to_pyiron = []
+        self._idx_pyiron_to_user = []
+        
+    @property
+    def idx_user_to_pyiron(self):
+        if self.structure is None:
+            return None
+        if len(self._idx_user_to_pyiron) == 0:
+            self._map_pyiron_to_user_idx()
+        return self._idx_user_to_pyiron
+
+    @property
+    def idx_pyiron_to_user(self):
+        if self.structure is None:
+            return None
+        if len(self._idx_pyiron_to_user) == 0:
+            self._map_pyiron_to_user_idx()
+        return self._idx_pyiron_to_user
```

Comment:
These are not strictly necessary, right? Everything that can access those could also access `job.input.idx_pyiron_to_user` if I see it correctly?

Reply:
I was just too lazy and copy pasted from the older pull - I still need to sort out the failing test and rename the property as per our discussion. It will probably become really long, since it's public and we want to avoid the nameclash with the chemical indices. I still think it's kind of annoying that we have to avoid that name in the first place. I think chemical species "indices" should give way to structure indices

Reply:
Can you show me why you think it is redundant?

Reply:
I don't usually write code that has this complex logic, so I may have missed an easier way to achieve what is being done here, happy to learn.